### PR TITLE
Add dependabot config

### PR DESCRIPTION
**Pull request by cdesiniotis**

### CODE REVIEW

**Review by elezar — APPROVED** [review]
LGTM.

I squashed your last two commits (DCO was failing) and changed the order of the dependabot entries.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Please see the documentation for all configuration options:
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "gomod"
```
Comment (elezar) [inline .github/dependabot.yml:6]: @cdesiniotis one thing to consider is whether we want update for the latest `release-*` branch too?

**Comment by cdesiniotis** [comment]
Thank you @elezar!

File: .github/dependabot.yml (hunk 1 above)
Comment (cdesiniotis) [reply .github/dependabot.yml:6]: Not opposed to it. @tariq1890 what do you think?

**Pull request merged**